tests/ashley/api/test_api_upload_image.py

```diff
@@ -0,0 +1,203 @@
+"""Tests to upload image API."""
+import datetime
+import json
+import urllib.request
+from io import BytesIO
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.files.base import ContentFile
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.test import TestCase
+from django.test.utils import override_settings
+from machina.core.db.models import get_model
+from PIL import Image
+
+from ashley.factories import ForumFactory, UploadImageFactory, UserFactory
+
+User = get_user_model()
+UploadImage = get_model("ashley", "UploadImage")
+
+
+class ImageUploadApiTest(TestCase):
```

Comment:
I'd add 3 tests : 

- Check that a user cannot delete any uploaded file
- Check that a user cannot update any uploaded file
- Check that if a user uploads 2 files with the same filename, in the same course, they have distinct urls (you handled this case but i don't see any test to confirm it)